editor/scene_tree_dock.cpp

```diff
@@ -493,16 +493,25 @@ void SceneTreeDock::_tool_selected(int p_tool, bool p_confirm_override) {
 
 				// Preserve ownership relations ready for pasting.
 				List<Node *> owned;
-				node->get_owned_by(node->get_owner() ? node->get_owner() : node, &owned);
+				Node *owner = node;
+				while (owner) {
+					List<Node *> cur_owned;
+					node->get_owned_by(owner, &cur_owned);
+					owner = owner->get_owner();
+					for (Node *F : cur_owned) {
+						owned.push_back(F);
+					}
+				}
 
 				for (Node *F : owned) {
 					if (!duplimap.has(F) || F == node) {
 						continue;
 					}
 					Node *d = duplimap[F];
```

Comment:
```suggestion
					if (F == node) {
						continue;
					}

					HashMap<const Node *, Node *>::Iterator G = duplimap.find(F);
					if (!G) {
						continue;
					}
					Node *d = G->value;
```
Unrelated to this PR, but this code can be improved.